.github/workflows/scripts/push-images.sh

```diff
@@ -1,20 +1,66 @@
-#!/bin/bash
+#!/usr/bin/env bash
+# shellcheck shell=bash
+##
+## USAGE: __PROG__
+##
+## "__PROG__" publishes oci tarballs created with xbuild to a registry.
+##
+## Usage example(s):
+##   ./__PROG__ 1.5.2
+##   ./__PROG__ v1.5.2
+##   ./__PROG__ v1.5.2 -scratch
+##   ./__PROG__ refs/tags/v1.5.2
+##   ./__PROG__ refs/tags/v1.5.2 -scratch
+##
+## Commands
+## - ./__PROG__ <version> [image-variant]   pushes images to the registry using given version.
 
 set -e
 
-IMAGETAG="$1"
-if [ -z "$IMAGETAG" ]; then
-    echo "IMAGETAG not provided!" 1>&2
-    echo "Usage: push-images.sh IMAGETAG" 1>&2
-    exit 1
+function usage {
+  grep '^##' "$0" | sed -e 's/^##//' -e "s/__PROG__/$me/" >&2
+}
+
+me=$(basename "$0")
+
+version="$1"
+if [ -z "$version" ]; then
+  usage
+  echo -e "\n Errors:\n * the version must be provided."
+  exit 1
 fi
 
-echo "Pushing images tagged as $IMAGETAG..."
+# remove the git tag prefix
+# Push the images using the version tag (without the "v" prefix).
+# Also strips the refs/tags part if the GITHUB_REF variable is used.
+version="${version#refs/tags/v}"
+version="${version#v}"
+
+variant="$2"
+if [ -n "$variant" ] && [ "$variant" != "-scratch" ]; then
+  usage
+  echo -e "\n Errors:\n * The only supported variant is '-scratch'."
+  exit 1
+fi
+
+OCI_IMAGES=(
+  spire-server spire-agent k8s-workload-registrar oidc-discovery-provider
+)
+
+registry=gcr.io/spiffe-io
+if [ "$variant" = "-scratch" ] ; then
+  org_name=$(echo "$GITHUB_REPOSITORY" | tr '/' "\n" | head -1 | tr -d "\n")
+  org_name="${org_name:-spiffe}" # default to spiffe in case ran on local
+  registry=ghcr.io/$org_name
+fi
 
-for img in spire-server spire-agent k8s-workload-registrar oidc-discovery-provider; do
-    gcrimg=gcr.io/spiffe-io/"$img":"${IMAGETAG}"
-    echo "Executing: docker tag $img:latest-local $gcrimg"
-    docker tag "$img":latest-local "$gcrimg"
-    echo "Executing: docker push $gcrimg"
-    docker push "$gcrimg"
+echo "Pushing images ${OCI_IMAGES[*]} to ${registry} with tag ${version}".
+for img in "${OCI_IMAGES[@]}"; do
+  image_variant="${img}${variant}"
+  image_to_push="$registry/$img:${version}"
+  if { [ "$variant" = "-scratch" ] || [ "$variant" = "-static" ]; } && [ "$img" == "oidc-discovery-provider" ] ; then
```

Comment:
Applies the change in image name when pushing oidc-discovery-provider to ghcr.io. To equal the naming from the push-scratch-images script.

Reply:
maybe I'm missing something but in 51l you are validating that the only supported variant is `scratch`, but here expecting a `static`